internal/pkg/vm/actor/builtin/account/account.go

```diff
@@ -39,14 +53,58 @@ func UpgradeActor(a *actor.Actor) error {
 var _ dispatch.ExecutableActor = (*Actor)(nil)
 
 // signatures are the publicly (externally callable) methods of the AccountActor.
-var signatures = dispatch.Exports{}
+var signatures = dispatch.Exports{
+	Constructor: &dispatch.FunctionSignature{
+		Params: []abi.Type{abi.Address},
+		Return: []abi.Type{},
+	},
+}
 
 // Method returns method definition for a given method id.
-func (*Actor) Method(id types.MethodID) (dispatch.Method, *dispatch.FunctionSignature, bool) {
+func (a *Actor) Method(id types.MethodID) (dispatch.Method, *dispatch.FunctionSignature, bool) {
+	switch id {
+	case Constructor:
+		return reflect.ValueOf((*Impl)(a).Constructor), signatures[Constructor], true
+	}
 	return nil, nil, false
 }
 
 // InitializeState for account actors does nothing.
-func (*Actor) InitializeState(_ runtime.Storage, _ interface{}) error {
-	return nil
+func (*Actor) InitializeState(storage runtime.Storage,  initializerData interface{}) error {
+	state, ok := initializerData.(*State)
+	if !ok {
+		return errors.NewFaultError("Initial state to account actor is not a account.State struct")
+	}
+
+	if state.Address.Protocol() != address.SECP256K1 && state.Address.Protocol() != address.BLS {
```

Comment:
It seems off that we create new account actors with `NewActor` that don't satisfy this condition in consensus/genesis.  Their states will always need to be uninitialized.  It seems like these things shouldn't even be considered account actors.

Reply:
The network actor is the only Account actor that isn't derived from a public key address. It is added to the state tree directly. This condition should hold for all other actors.